From e603674c97e2b9654f7a599e33d3d7f7f92d8f5d Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:45 -0400
Subject: [fs] nfsd: Always use lookup_clientid() in nfsd4_process_open1

Message-id: <1410898849-22126-42-git-send-email-bfields@redhat.com>
Patchwork-id: 95798
O-Subject: [Patch RHEL7 041/165] nfsd: Always use lookup_clientid() in nfsd4_process_open1
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

In later patches, we'll be moving the stateowner table into the
nfs4_client, and by doing this we ensure that we have a cached
nfs4_client pointer.

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 2d91e8953cb046d9eef281ddc608fee31a942f35
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 3428819..f82aec4 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -3050,19 +3050,19 @@ nfsd4_process_open1(struct nfsd4_compound_state *cstate,
  if (open->op_file == NULL)
   return nfserr_jukebox;
 
+ status = lookup_clientid(clientid, cstate, nn);
+ if (status)
+  return status;
+ clp = cstate->clp;
+
  strhashval = ownerstr_hashval(clientid->cl_id, &open->op_owner);
  oo = find_openstateowner_str(strhashval, open, cstate->minorversion, nn);
  open->op_openowner = oo;
  if (!oo) {
-  status = lookup_clientid(clientid, cstate, nn);
-  if (status)
-   return status;
-  clp = cstate->clp;
   goto new_owner;
  }
  if (!(oo->oo_flags & NFS4_OO_CONFIRMED)) {
   /* Replace unconfirmed owners without checking for replay. */
-  clp = oo->oo_owner.so_client;
   release_openowner(oo);
   open->op_openowner = NULL;
   goto new_owner;
@@ -3070,7 +3070,6 @@ nfsd4_process_open1(struct nfsd4_compound_state *cstate,
  status = nfsd4_check_seqid(cstate, &oo->oo_owner, open->op_seqid);
  if (status)
   return status;
- clp = oo->oo_owner.so_client;
  goto alloc_stateid;
 new_owner:
  oo = alloc_init_open_stateowner(strhashval, open, cstate);
-- 
1.7.1